pkg/skaffold/runner/generate_pipeline.go

```diff
@@ -17,66 +17,52 @@ limitations under the License.
 package runner
 
 import (
-	"bufio"
 	"bytes"
 	"context"
 	"encoding/json"
-	"fmt"
 	"io"
 	"io/ioutil"
-	"os"
-	"os/exec"
-	"strings"
-
-	"github.com/ghodss/yaml"
-	"github.com/pkg/errors"
 
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/color"
-	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/pipeline"
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest"
-	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/version"
+	"github.com/ghodss/yaml"
+	"github.com/pkg/errors"
 
+	pipeline "github.com/GoogleContainerTools/skaffold/pkg/skaffold/generate_pipeline"
 	tekton "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
-	yamlv2 "gopkg.in/yaml.v2"
-	corev1 "k8s.io/api/core/v1"
-)
-
-var (
-	// for testing
-	reader = bufio.NewReader(os.Stdin)
 )
 
 func (r *SkaffoldRunner) GeneratePipeline(ctx context.Context, out io.Writer, config *latest.SkaffoldConfig, fileOut string) error {
-	err := createSkaffoldProfile(out, config, r.runCtx.Opts.ConfigurationFile)
+	err := pipeline.CreateSkaffoldProfile(out, config, r.runCtx.Opts.ConfigurationFile)
```

Comment:
since most of this function is calling other functions in the `pipeline` package, I think most of `GeneratePipeline` could also be moved into the `pipeline` pkg, and this function could handle:

1. getting the pipeline contents
2. writing to the file